### PR TITLE
Feat/bar

### PR DESCRIPTION
**Pull request by pearmini**
Render charts related to interval geometry(e.g. bar, column, pie, stack column), the chart is rendered with components but without animation and interaction.(**Test coverage：95%**)

- **transform**: fetch, sortBy
- **encode**: constant, value, field
- **infer**: maybeTuple, maybeZeroX1, maybeZeroY2
- **statistic**: stackY, dodgeX
- **scale**: identity, band, linear, ordinal
- **palette**: category10, category20
- **coordinate**: cartesian, polar, transpose
- **component**: axisX, axisY, legendContinuous, legendCategory
- **geometry**: interval
- **shape**: rect, hollowRect
- **theme**: light
- **runtime**: Plot a view without animation and interaction.

**Basic Usage:**

```js
import { render } from '@antv/g2';

const chat = render({
  type: 'interval',
  data: [
    { genre: 'Sports', sold: 275 },
    { genre: 'Strategy', sold: 115 },
    { genre: 'Action', sold: 120 },
    { genre: 'Shooter', sold: 350 },
    { genre: 'Other', sold: 150 },
  ],
  encode: {
    x: 'genre',
    y: 'sold',
    color: 'genre',
  },
});

document.body.appendChild(chart);
```
**Render with callback.**
```js
import { render } from '@antv/g2';

const chat = render({
  type: "interval",
  transform: [
    {
      type: "fetch",
      url: "https://gw.alipayobjects.com/os/bmw-prod/ce45e3d7-ba78-4a08-b411-28df40ef9b7f.json",
    },
  ],
  encode: {
    x: "genre",
    y: "sold",
  },
}, {}, () => console.log('render successfully');

document.body.appendChild(chart);
```
**Custom component.**(shape for example).

```js
import { Polygon } from '@anv/g';

const options = {
  type: 'interval',
  data: [
    { genre: 'Sports', sold: 275 },
    { genre: 'Strategy', sold: 115 },
    { genre: 'Action', sold: 120 },
    { genre: 'Shooter', sold: 350 },
    { genre: 'Other', sold: 150 },
  ],
  encode: {
    x: 'genre',
    y: 'sold',
    shape: {
      type: 'constant',
      // Inline custom.
      value: () => {
        return (P, style) => {
          const [p0, p1, p2, p3] = P;
          const pm = [(p0[0] + p1[0]) / 2, p0[1]];
          const { color } = style;
          return new Polygon({
            style: {
              ...style,
              fill: color,
              points: [pm, p2, p3],
            },
          });
        };
      },
    },
  },
};
```
```js
import { render, context } from '@antv/g2';

const TriangleShape = () => {
  return (P, style, coordinate) => {
    const [p0, p1, p2, p3] = P;
    const pm: Vector2 = [(p0[0] + p1[0]) / 2, p0[1]];
    const { color } = style;
    return new Polygon({
      style: {
        ...style,
        fill: color,
        points: [pm, p2, p3],
      },
    });
  };
};

const context = {
  // Global Custom.
  library: Object.assign(createLibrary(), {
    'shape.triangle': TriangleShape,
  }),
};

const chart = render(
  {
    type: 'interval',
    data: [
      { genre: 'Sports', sold: 275 },
      { genre: 'Strategy', sold: 115 },
      { genre: 'Action', sold: 120 },
      { genre: 'Shooter', sold: 350 },
      { genre: 'Other', sold: 150 },
    ],
    encode: {
      x: 'genre',
      y: 'sold',
      shape: 'triangle',
    },
  },
  context,
);

document.body.appendChild(chart);
```

### CODE REVIEW

**Review by hustcc — APPROVED** [review]

File: src/runtime/types/common.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+export type G2Theme = {
+  defaultColor?: string;
```
Comment (visiky) [inline src/runtime/types/common.ts:2]: 这个应该是临时的？我后续可以修改是吧？

File: src/runtime/types/component.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  | Encode
+  | Infer
+  | Statistic
+  | Scale
+  | CoordinateTransform
+  | Palette
+  | Mark
+  | Shape
+  | Theme
+  | GuideComponent;
```
Comment (visiky) [inline src/runtime/types/component.ts:56]: 其实我也在纠结，要不要用 Guide 代表辅助组件就好了。GUI 不如改名为 guides 就好了。图形元素也是 component，万物都是

File: src/runtime/types/common.ts (hunk 1 above)
Comment (pearmini) [reply src/runtime/types/common.ts:2]: 是的

File: src/runtime/types/component.ts (hunk 2 above)
Comment (pearmini) [reply src/runtime/types/component.ts:56]: 这个地方命名后面还是要统一改一下，暂时用 GuideComponent

**Review by visiky — APPROVED** [review]
👏

**Pull request merged**